### PR TITLE
Create separate packages for interface and compile using lerna

### PR DESCRIPTION
**Pull request by majapw**
@ljharb @lelandrichardson @airbnb/webinfra 

Okay so this is a bit of a raging trashfire.
a) I could not figure out (conceptually, in my head) how to make this a patch change rather than a breaking change. I support I could make a third package named react-with-styles-interface-css` that exports the other two for now and name the interface package something else... does that seem reasonable?
b) I am not quite sure where the version scripts belong... do both packages need them? should they remain in the top folder? 

Please halp.

### CODE REVIEW

**Comment by majapw** [comment]
Yay tests pass! :D

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,1762 @@
+{
```
Comment (lencioni) [inline package-lock.json]: I think we don't want to be adding this package.lock file. Can fix like: https://github.com/airbnb/react-with-styles/commit/5175f997e40918509fd3fb6a2ba9f34311481eba

**Comment by lencioni** [comment]
One comment about package-lock, otherwise LGTM

File: package-lock.json (hunk 1 above)
Comment (ljharb) [reply package-lock.json]: thanks, will fix

**Comment by ljharb** [comment]
The tests seem to be failing because the compiler requires a not-yet-published version of the interface.

**Pull request merged**